addon-test-support/assertions/expect-deprecation.js

```diff
@@ -0,0 +1,37 @@
+import {
+  getDeprecationsDuringCallback,
+  getDeprecations,
+} from '@ember/test-helpers';
+import checkMatcher from './utils/check-matcher';
+
+export default function expectDeprecation(cb, matcher) {
```

Comment:
read and consider https://github.com/emberjs/ember.js/blob/master/packages/internal-test-helpers/lib/ember-dev/deprecation.ts